test/preview/components/builders/test_answers_builders.py

```diff
@@ -0,0 +1,156 @@
+            component.run(queries=["test query"], replies=[["Answer: AnswerString, AnswerString2"]], metadata=[[{}]])
+
+    def test_run_with_pattern_set_at_runtime(self):
+        component = AnswersBuilder()
+        answers = component.run(
+            queries=["test query"], replies=[["Answer: AnswerString"]], metadata=[[{}]], pattern=r"Answer: (.*)"
+        )
+        assert len(answers) == 1
+        assert len(answers[0]) == 1
+        assert answers[0][0].data == "AnswerString"
+        assert answers[0][0].metadata == {}
+        assert answers[0][0].query == "test query"
+        assert answers[0][0].documents == []
+        assert isinstance(answers[0][0], GeneratedAnswer)
+
+    def test_run_with_documents_without_reference_pattern(self):
+        component = AnswersBuilder()
+        answers = component.run(
+            queries=["test query"],
+            replies=[["Answer: AnswerString"]],
+            metadata=[[{}]],
+            documents=[[Document(content="test doc 1"), Document(content="test doc 2")]],
+        )
+        assert len(answers) == 1
+        assert len(answers[0]) == 1
+        assert answers[0][0].data == "Answer: AnswerString"
+        assert answers[0][0].metadata == {}
+        assert answers[0][0].query == "test query"
+        assert len(answers[0][0].documents) == 2
+        assert answers[0][0].documents[0].content == "test doc 1"
+        assert answers[0][0].documents[1].content == "test doc 2"
+
+    def test_run_with_documents_with_reference_pattern(self):
+        component = AnswersBuilder(reference_pattern="\\[(\\d+)\\]")
+        answers = component.run(
+            queries=["test query"],
+            replies=[["Answer: AnswerString[2]"]],
+            metadata=[[{}]],
+            documents=[[Document(content="test doc 1"), Document(content="test doc 2")]],
+        )
+        assert len(answers) == 1
+        assert len(answers[0]) == 1
+        assert answers[0][0].data == "Answer: AnswerString[2]"
+        assert answers[0][0].metadata == {}
+        assert answers[0][0].query == "test query"
+        assert len(answers[0][0].documents) == 1
+        assert answers[0][0].documents[0].content == "test doc 2"
+
+    def test_run_with_documents_with_reference_pattern_and_no_match(self):
+        component = AnswersBuilder(reference_pattern="\\[(\\d+)\\]")
+        answers = component.run(
+            queries=["test query"],
+            replies=[["Answer: AnswerString[3]"]],
+            metadata=[[{}]],
+            documents=[[Document(content="test doc 1"), Document(content="test doc 2")]],
+        )
+        assert len(answers) == 1
+        assert len(answers[0]) == 1
+        assert answers[0][0].data == "Answer: AnswerString[3]"
+        assert answers[0][0].metadata == {}
+        assert answers[0][0].query == "test query"
+        assert len(answers[0][0].documents) == 0
+
+    def test_run_with_reference_pattern_set_at_runtime(self):
+        component = AnswersBuilder()
```

Comment:
(as above) Nitpick: let's put another pattern here in the init to make sure AnswerBuilder uses the right one when both are set.